### PR TITLE
[CALCITE-1598] Initial Pig adapter work

### PR DESCRIPTION
**Pull request by elilevine**
@julianhyde, FYI. Would appreciate any comments you might have on this approach. Thanks.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -307,6 +314,18 @@ limitations under the License.
         <version>${httpcore.version}</version>
       </dependency>
       <dependency>
+        <groupId>org.apache.pig</groupId>
+        <artifactId>pig</artifactId>
+        <version>${pig.version}</version>
+        <classifier>h2</classifier>
```
Comment (elilevine) [inline pom.xml]: Pig for Hadoop 2

File: pom.xml [hunk 2]
```diff
@@ -74,7 +74,7 @@ limitations under the License.
     <guava.version>18.0</guava.version>
     <joda.version>2.8.1</joda.version>
     <h2.version>1.4.185</h2.version>
-    <hadoop.version>2.6.0</hadoop.version>
+    <hadoop.version>2.7.0</hadoop.version>
```
Comment (elilevine) [inline pom.xml]: Had to upgrade to 2.7 to work around a broken Guava dependency in 2.6. https://issues.apache.org/jira/browse/HADOOP-10101

**Comment by julianhyde** [comment]
@elilevine Is this PR ready for review? Do you believe it could go into release 1.12 as is?

**Comment by elilevine** [comment]
@julianhyde, this PR is in a good shape: tests and checkstyle checks pass. After your suggestion on the [Jira](https://issues.apache.org/jira/browse/CALCITE-1598?focusedCommentId=15840763&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15840763), I have added a _PigSchema_, _PigToEnumerableConverter_ and associated classes in order to support Calcite SQL by the Pig adapter. Test cases in _PigAdapterTest_ show that the flow from SQL to Pig-specific relational expression trees to Pig scripts is now supported.

The big piece that is currently missing is actually executing Pig scripts and returning results via Calcite. I am working on that now, initially supporting Pig's local execution mode. Getting close. I'll leave it up to you to make the call on whether or not you want to check it into version 1.12 without this functionality.

Other limitations of the existing Pig adapter impl:
- Only supports VARCHAR columns. Need more data type support.
- Filters are not pushed to be before joins. To be investigated.
- Need expansion of supported functionality and test coverage for more permutations of filters, aggregations and joins.

**Pull request closed without merging**